app/scripts/background.js

```diff
@@ -1015,9 +1015,13 @@ export function setupController(
 
   function getPendingApprovalCount() {
     try {
+      const unlockCount = Math.min(
+        controller.appStateController.waitingForUnlock.length,
+        1,
+      );
```

Comment:
Context: This `waitingForUnlock` is an array of promises from requests internally (any maybe externally) that require the wallet to be unlocked. As soon as the wallet is unlocked, all of these promises get resolved.

From a UI perspective, I don't see why we need to have more than 1 unlock count - especially as all the requests will get resolved once the wallet is unlocked.

Otherwise we can add a larger bound here. This will prevent the unlock requests count from growing (e.g. prevent 1000+ count on the confirmations badge)